core/controller/src/main/scala/org/apache/openwhisk/core/controller/Actions.scala

```diff
@@ -301,9 +301,9 @@ trait WhiskActionsApi extends WhiskCollectionAPI with PostActionActivation with
             // NOTE: response status will not exist in the JSON object if ?result == true
             // and instead clients must check if 'error' is in the JSON
             // PRESERVING OLD BEHAVIOR and will address defect in separate change
-            complete(BadGateway, response)
```

Comment:
Please look at the comment right above which should be amended.

Reply:
you mean `PRESERVING OLD BEHAVIOR and will address defect in separate change`, right?

Reply:
Yep with your change you've finally fixed the behavior :) so the comment no longer holds as is.